apps/src/weblab/brambleUtils.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Remove disallowed content from the HTML at the given path. The editor will be set as readOnly
+ * while this method overwrites the HTML file. Use this method carefully as it does overwrite
+ * file contents.
+ *
+ * @param {Object} fileSystem - The current file system. This method requires `readFile` and `writeFile` methods.
+ * @param {Object} brambleProxy - Communicate with Bramble. This method requires `enableReadonly` and `disableReadonly` methods.
+ * @param {string} path - Path to the HTML file.
+ * @param {Array<string>} disallowedTags - HTML tags that should be removed.
+ * @param {func} callback - Will be called with the given path. Optional.
+ */
+function removeDisallowedHtmlContent(
+  fileSystem,
+  brambleProxy,
+  path,
+  disallowedTags,
+  callback
+) {
+  function wrappedCallback() {
+    if (callback) {
+      callback(path);
+    }
+  }
+
+  if (!path.endsWith('.html')) {
+    wrappedCallback();
+    return;
+  }
+
+  fileSystem.readFile(path, 'utf8', function(error, data) {
+    if (error) {
+      wrappedCallback();
+      return;
+    }
+
+    const dom = new DOMParser().parseFromString(data, 'text/xml');
```

Comment:
@aaronwaggener continuing our discussion from [this thread](https://github.com/code-dot-org/code-dot-org/pull/39207#discussion_r582423503) here...

i found that parsing our HTML as 'text/xml' (instead of 'text/html') solves a couple of problems i was seeing before:
1. we can use XMLSerializer with minimal formatting changes (lines 56-58 below)
2. [a new problem i discovered after we chatted yesterday] this mimeType handles self-closing tags in the way we want:

```javascript
// using 'text/html' mimeType, <div> is considered a child of <script/> and is removed
// using 'text/xml' mimeType, <script/> is self-contained
<script/>
<div>please don't delete me</div>
```

i did some manual testing and this change didn't break any unit tests, but do you see any issues with using the XML mimeType here?

Reply:
yeah, i noticed that as well but didn't think it was an issue because it only affects empty elements and doesn't break the user's project or bramble execution. is that a reasonable assessment, or am i missing a deeper issue that this (changing `<p></p>` to `<p/>`) could cause?

Reply:
I was thinking the deeper issue is that our tool is outputting something that we would not want the student to be practicing/learning (invalid HTML markup). Potentially more problematic would be an empty `<head>` or `<body>` element as I think it could cause some confusion to see `<head/>` or `<body/>` in markup.

I am still a little worried that parsing arbitrary HTML with an XML parser and then serializing to string again will have some other unintended markup consequences but nothing else has come up in my quick tests in the console.

Reply:
> Potentially more problematic would be an empty <head> or <body> element as I think it could cause some confusion to see <head/> or <body/> in markup.
good point -- i hadn't noticed this because it only occurs if the tag is on a single line (`<head></head>` goes to `<head/>` but `<head>\n</head>` doesn't)

yeah, your overall concern with using an XML parser is valid. i couldn't figure out a way forward with self-closing tags when the string is parsed to HTML without also incorporating a regex or something similar, but i can investigate that further if that's the best path forward